test/speech/index.js

```diff
@@ -0,0 +1,836 @@
+          assert.ifError(err);
+          assert.deepEqual(response, [expected]);
+
+          done();
+        }
+      );
+    });
+
+    it('should return an error from recognize()', function(done) {
+      var error = new Error('Error.');
+      var RES = {};
+
+      speech.recognize = function(config, callback) {
+        callback(error, null, RES);
+      };
+
+      speech.detect(FILE, REQ, function(err, response, apiResponse) {
+        assert.strictEqual(err, error);
+        assert.strictEqual(response, null);
+        assert.strictEqual(apiResponse, RES);
+        done();
+      });
+    });
+
+    it('should return the resp from recognize()', function(done) {
+      var RES1 = new speech.protos.Speech.SyncRecognizeResponse({
+        results: [
+          {
+            alternatives: []
+          }
+        ]
+      });
+      var RES2 = {
+        results: [
+          {
+            alternatives: []
+          }
+        ]
+      };
+
+      speech.recognize = function(config, callback) {
+        callback(null, RES1, RES2);
+      };
+
+      speech.detect(FILE, REQ, function(err, response, apiResponse) {
+        assert.ifError(err);
+
+        assert.deepEqual(response, RES1.get('results')[0].get('alternatives'));
+        assert.strictEqual(apiResponse, RES2);
+
+        done();
+      });
+    });
+  });
+
+  describe('createRecognizeJob', function() {
+    var REQ = {};
+
+    it('should make the correct API request', function(done) {
+      var request = REQ;
+
+      speech.request = function(protoOpts, reqOpts) {
+        assert.strictEqual(protoOpts.service, 'Speech');
+        assert.strictEqual(protoOpts.method, 'asyncRecognize');
+        assert.strictEqual(reqOpts, REQ);
+
+        done();
+      };
+
+      speech.createRecognizeJob(request, assert.ifError);
+    });
+
+    describe('error', function() {
+      var error = new Error('Error.');
+      var RES = {};
+
+      beforeEach(function() {
+        speech.request = function(protoOpts, reqOpts, callback) {
+          callback(error, RES);
+        };
+      });
+
+      it('should execute callback with error & API response', function(done) {
+        speech.createRecognizeJob(REQ, function(err, operation, apiResponse) {
+          assert.strictEqual(err, error);
+          assert.strictEqual(operation, null);
+          assert.strictEqual(apiResponse, RES);
+          done();
+        });
+      });
+    });
+
+    describe('success', function() {
+      var RES = {
+        name: '1234'
+      };
+
+      beforeEach(function() {
+        speech.request = function(protoOpts, reqOpts, callback) {
+          callback(null, RES);
+        };
+      });
+
+      it('should execute callback with results & API resp', function(done) {
+        speech.createRecognizeJob(REQ, function(err, operation, apiResponse) {
+          assert.ifError(err);
+
+          assert.equal(operation.name, RES.name);
+          assert.equal(operation.results, undefined);
```

Comment:
I think we're going to want to stub `Speech#operation` and ensure that whatever that returns is what is passed into the callback.